master/internal/api_master.go

```diff
@@ -113,24 +118,42 @@ func (a *apiServer) PatchMasterConfig(
 			if !isValidLogLevel(req.Config.Log.Level) {
 				panic(fmt.Sprintf("invalid log level: %s", req.Config.Log.Level))
 			}
-			a.m.config.Log.Level = req.Config.Log.Level
-			a.m.config.Log.Color = req.Config.Log.Color
-			logger.SetLogrus(a.m.config.Log)
+			oldConfig := a.m.config.Load()
+			newConfig := *oldConfig
+			newConfig.Log.Level = req.Config.Log.Level
+			newConfig.Log.Color = req.Config.Log.Color
```

Comment:
i think this works for now because it's values all the way down to the logger config but the shallow copy `*oldConfig` does won't be sufficient when we start changing stuff that is behind a pointer. we'll need a deep copy method.

Reply:
Done